### PR TITLE
New UI implementation (Login, Signup, Account Page)

### PR DESCRIPTION
**Pull request by 4855062503**
Continuing to update this PR, and fix Build error

### CODE REVIEW

**Comment by djpnewton** [comment]
you have conflicts because you (*again*) did not start a new branch and have commits in this branch that were already squashed and applied in a previous PR

**Comment by 4855062503** [comment]
> new branch and have commits in this branch that were already squashed and applied in a previous PR

Will fix this.

File: pubspec.yaml [hunk 1]
```diff
@@ -74,17 +74,11 @@ flutter:
...
-  #       - asset: fonts/Schyler-Regular.ttf
-  #       - asset: fonts/Schyler-Italic.ttf
-  #         style: italic
-  #   - family: Trajan Pro
-  #     fonts:
-  #       - asset: fonts/TrajanPro.ttf
-  #       - asset: fonts/TrajanPro_Bold.ttf
-  #         weight: 700
-  #
+fonts:
```
Comment (djpnewton) [inline pubspec.yaml:77]: remove this font, we will just use Poppins (also remove the added font file)

File: lib/main.dart [hunk 2]
```diff
@@ -51,11 +53,13 @@ class MyApp extends StatelessWidget {
     initConfig();
     var theme = ThemeData(
       useMaterial3: false,
+      fontFamily: 'SF-Pro',
```
Comment (djpnewton) [inline lib/main.dart:56]: remove this line

File: lib/main.dart [hunk 3]
```diff
@@ -51,11 +53,13 @@ class MyApp extends StatelessWidget {
...
       brightness: ZapBrightness,
       scaffoldBackgroundColor: ZapBackground,
       appBarTheme: AppBarTheme(centerTitle: true, color: ZapSecondary),
       // ignore: deprecated_member_use
       accentColor: ZapPrimary,
+      colorScheme: ColorScheme.dark(primary: Color(0xFF1B1C25)),
```
Comment (djpnewton) [inline lib/main.dart:62]: remove this I have already set the basic theme colors on master

File: lib/main.dart [hunk 4]
```diff
@@ -287,17 +295,25 @@ class _MyHomePageState extends State<MyHomePage> with WidgetsBindingObserver {
   }
 
   Future<void> _login() async {
-    AccountLogin? login;
+    dynamic login;
```
Comment (djpnewton) [inline lib/main.dart:298]: dont use dynamic make a custom `@freezed` class

**Review by djpnewton — CHANGES_REQUESTED** [review]
use the freezed libary to avoid using `dynamic` (see beryllium.dart)

remove the SFpro font (font already set to poppins on master)

rename accountForm.dart to account_form.dart like other files in the project

**Comment by 4855062503** [comment]
Still working on the Freezed stuff. Hopefully implemented by tonight.

File: pubspec.yaml [hunk 5]
```diff
@@ -74,17 +74,5 @@ flutter:
   # "family" key with the font family name, and a "fonts" key with a
   # list giving the asset and other descriptors for the font. For
   # example:
-  # fonts:
```
Comment (djpnewton) [inline pubspec.yaml:77]: remove unecesary changes

File: web/index.html [hunk 6]
```diff
@@ -95,3 +95,4 @@
   </div>
 </body>
 </html>
+
```
Comment (djpnewton) [inline web/index.html:98]: unessesary change

**Review by djpnewton — CHANGES_REQUESTED** [review]
-restore old logo up top (add new logo for the login pages only)
- remove font file

**Comment by djpnewton** [comment]
When loading with invalid auth I end up on this page (note the "reset" button looks odd):
image
image

**Comment by djpnewton** [comment]
merged from master

**Comment by djpnewton** [comment]
- [ ] lets make the signup page look closer to the original brief (remove the profile image widget, add in the large logo etc)
- [ ] remove the lost password from first screen (only have it on the login screen)
- [ ] move the "hey there lets get started" from the first screen and onto the register screen (like the design brief)
- [ ] add the "----- Or -----" from the design brief onto the login and register screens
- [ ] make the grey buttons the same grey (some or different shades of grey)

**Comment by djpnewton** [comment]
couple of issues 

- on pixel 5:
image


- initial page says "Register" but other pages say "Create an Account", lets use "Create an Account" everywhere

**Comment by djpnewton** [comment]
@4855062503 I tidied up the code a bit see the latest commit

-move everything related to login/registration to `login.dart`
-move some more repeated widget code to `widgets.dart`
-changed logic flow to use two new freezed enums (`LoginChoice` and `LoginResult`) 
this way we can use recursion to most of the descsions about what to do next in `_processLoginChoice()`

NB. still want to get rid of the `BeforeLoginForm_KillMe` screen and go strait to the register screen

**Comment by djpnewton** [comment]
I removed the changes to exchange.dart and withdrawal.dart, I think these should go in a separate PR/branch:
[exchange_withdrawal.diff.txt](https://github.com/zap-me/alloy/files/8885475/exchange_withdrawal.diff.txt)

@4855062503 
There is only one thing left to do I think here is how can we go directly to the account creation screen and get rid of the `BeforeLoginForm_KillMe` screen. 

Once that is done you just need to test the following scenarios with and without 2FA:

- [ ] login
- [ ] login failure (invalid credentials, ie use the reset button)
- [ ] login failure (network error, fix network and retry)
- [ ] register
- [ ] recover account with email

**Comment by djpnewton** [comment]
ok this is ready for testing now

**Comment by 4855062503** [comment]
@djpnewton 
✅ = test went as expected
> login w/out 2FA ✅
> signup w/out matching passwords ✅ (note: form validation text is not centered)
> signup w/ weak password ❔ - get "authorization error"
> signup w/ strong password & new email ❌  - get "authorization error"
> login w/ incorrect credentials ✅
> signup with existing email ✅ (note: could tell user that the email exists instead of throwing 'authorization error')
> login failure (invalid credentials, ie use the reset button) ✅
>  login failure (network error, fix network and retry) ✅ (note: no offline service workers exist, so upon a page refresh, get dinosaur Chrome network page. Some assets fail to load without network, if staying in the app long enough, network error alert pops up notifying user)
> recover account w/ email ✅ (note: permissions modal still pops up when link clicked)
> try recover account w/ nonemail (e.g. enter invalid input) ✅ (fails)
> try recover account with legit email which doesn't exist ❌ - doesn't warn user that their email doesn't exist, just has spinning modal waiting for confirmation

**Pull request merged**